elbepack/initvmaction.py

```diff
@@ -148,6 +152,54 @@ def initvm_state(self):
         return self.initvm.info()[0]
 
 
+def is_qemu_running():
+    """
+    Test if QEMU vm is running.
+
+    This function is used in QEMU mode to test if the initvm is running.
+    It's using lsof to see if a process on the current machine has bound to the
+    soap port. If such a process is found, it's expected to be the initvm.
+    """
+
+    # test is lsof command is available, and fail with error exit code if not.
+    try:
+        system('which lsof > /dev/null 2>&1')
+    except CommandError:
+        print(f'The command "lsof" is required.\nPlease install lsof: "sudo apt install lsof"')
+        sys.exit(209)
+    
+    # test if a process is listening on the configured soap port.
+    port = cfg['soapport']
+    try:
+        system(f'lsof -i tcp:{port} | grep LISTEN > /dev/null 2>&1')
+    except CommandError:
+        print(f'No process bound to port {port}.')
+        return False
+    # grep was successful - there is a process
+    return True
+
+def test_soap_communication(sleep=10):
+    """
+    Test communication with soap service.
+
+    In case of error, this fuction terminates the command with exit code 123.
+
+    This function tests the soap service communication by requesting the
+    list of projects. If this works, the communication is ok and the service
+    is up and seems to be healty.
+    """
+    # test if we can communicate with the soap service
+    stop = time.time() + 300
+    while True:
```

Comment:
We really should have some helper function to implement waiting.
It doesn't need to come from this PR tough.